ProductService/src/main/java/com/metaphorce/exam/ecommerce/product/controller/CategoryController.java

```diff
@@ -0,0 +1,138 @@
+
+import jakarta.validation.Valid;
+import lombok.extern.slf4j.Slf4j;
+
+@RequestMapping("/category")
+@RestController
+@Slf4j
+public class CategoryController {
+	@Autowired
+	private CategoryService service;
+	
+	/**
+	 * Implements the HTTP Post to create a Category
+	 * @param Category Data to insert
+	 * @return New Category create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@PostMapping
+	public ResponseEntity<CategoryDTO> createCategory(@Valid @RequestBody CategoryDTO CategoryRequest) {
+		log.info("Category to create: " + CategoryRequest);
+		ResponseEntity<CategoryDTO> response = null;
+		Category newCategory = this.service.create(CategoryMapper.mapper.toEntity(CategoryRequest));
+
+		if(newCategory != null) {
+			response = ResponseEntity.ok(CategoryMapper.mapper.toDTO(newCategory));
+			log.info("Category created: " + newCategory);
+		} else {
+			response = ResponseEntity.badRequest().build();
+			log.info("Error to create Category: " + CategoryRequest);
+		}
+
+		return response;
+	}
+
+	/**
+	 * Implements the HTTP Get to find all Categorys
+	 * @return List of Categorys
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@GetMapping
+	public ResponseEntity<List<CategoryDTO>> getAllCategories() {
+		log.info("Get All Categorys ...");
+		ResponseEntity<List<CategoryDTO>> response = null;
+		List<Category> listCategorys = this.service.getAll();
+
+		if(listCategorys != null) {
+			log.info("Categorys[" + listCategorys.size() + "]");
+			response = ResponseEntity.ok(listCategorys.stream().map(CategoryMapper.mapper::toDTO).toList());
+		} else {
+			response = ResponseEntity.notFound().build();
+			log.info("Error to get all Categorys");
+		}
+
+		return response;
+	}
+
+	/**
+	 * Implements HTTP Get to find a Category by id
+	 * @param id Category Id
+	 * @return Category data
+	 * @return New Category create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024
+	 */
+	@GetMapping("/{id}")
+	public ResponseEntity<CategoryDTO> getCategory(@PathVariable Integer id) {
+		log.info("Get Category[" + id + "]");
+		return service.get(id).map((Category Category) -> {
+			log.info("Category found: " + Category);
+			return ResponseEntity.ok(CategoryMapper.mapper.toDTO(Category));
+		}).orElseGet(() -> {
+			log.warn("Category not found: " + id);
+			return ResponseEntity.notFound().build();
+		});
+	}
+
+	/**
+	 * Implements HTTP Put for update a Category
+	 * @param id Category Id
+	 * @param Category Category data to update
+	 * @return Category modified
+	 * @return New Category create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024
+	 */
+	@PutMapping("/{id}")
+	public ResponseEntity<CategoryDTO> updateCategory(@PathVariable Integer id, @Valid @RequestBody CategoryDTO Category) {
+		log.info("Category[" + id + "] to update: " + Category);
+		ResponseEntity<CategoryDTO> response = null;
+		Category updateCategoryData = this.service.update(id, CategoryMapper.mapper.toEntity(Category));
+
+		if(updateCategoryData != null) {
```

Comment:
I don't understand why check if the value is null. It would be better to throw exceptions in the service when it doesn't find a certain resource; also, add a global exception handler.

In the case of not being able to create a resource, I think it's better to let a 500 status be generated because it probably means there's something wrong with the system.